### ISSUE TITLE
Not perfectly compatible with stylelint

### ISSUE THREAD

**Issue by FranzSkuffka** [seaneking/poststylus#13]
When I produce a stylelint warning i get this:
```
css/style.styl
499:19  ⚠  Unexpected !important (declaration-no-important) [stylelint]

Warning {
  type: 'warning',
  text: 'Unexpected !important (declaration-no-important)',
  line: 499,
  column: 19,
  severity: 1,
  rule: 'declaration-no-important',
  node: 
   Declaration {
     raws: { before: '\n  ', between: ': ' },
     type: 'decl',
     parent: 
      Rule {
        raws: [Object],
        type: 'rule',
        nodes: [Object],
        parent: [Object],
        source: [Object],
        selector: 'p',
        lastEach: 73,
        indexes: {} },
     source: { start: [Object], input: [Object], end: [Object] },
     prop: 'color',
     important: true,
     value: '#d13f19' },
  word: 'important',
  plugin: 'stylelint' } 0 [ Warning {
    type: 'warning',
    text: 'Unexpected !important (declaration-no-important)',
    line: 499,
    column: 19,
    severity: 1,
    rule: 'declaration-no-important',
    node: 
     Declaration {
       raws: [Object],
       type: 'decl',
       parent: [Object],
       source: [Object],
       prop: 'color',
       important: true,
       value: '#d13f19' },
    word: 'important',
    plugin: 'stylelint' } ]
[12:58:19] Finished 'stylus' after 742 ms
```
instead of just this:
```
css/style.styl
499:19  ⚠  Unexpected !important (declaration-no-important) [stylelint]

**Comment by jamime** [comment]
This is intended behavior. You'll be able to override this when #12 gets merged.

**Comment by FranzSkuffka** [comment]
Neat! Thanks for the heads up.

**Comment by seaneking** [comment]
derp, ignore that commit reference, wrong issue.

**Comment by nuragic** [comment]
Hi guys! Could you please show me your WebPack config using PostStylus and Stylelint? It would be really great... I'm stuck on it :sweat_smile: Thanks!

**Comment by FranzSkuffka** [comment]
Sorry, only used it with gulp so far. Maybe this helps though:
https://github.com/FranzSkuffka/gulp-stylus-pipeline/blob/master/lib/index.coffee

**Comment by nuragic** [comment]
Hi @FranzSkuffka thanks for your help! However there is one thing that is not still clear to me... are you linting the Stylus source or the processed CSS? Perhaps it's a generic question related to PostStylus I guess... am I wrong? @seaneking

**Comment by FranzSkuffka** [comment]
TL;DR: Processed CSS is linted.

Oh damn! I am sorry! I just realised I am not actually using poststylus in this particular code example :sob: but rather stylus & postcss.

In fact, your question is kind of post-stylus independent. I never tried linting stylus code with postCSS, not sure if it is possible, but I am always linting the resulting CSS.

Actually,  you do not need so much linting with stylus as it outputs well and consistently formatted code already. I used it in this particular example to make my dev-team adhere to BEM and get some structure into this CSS-mess.
https://github.com/FranzSkuffka/gulp-stylus-pipeline/blob/master/lib/exampleConfig.coffee
You should be aware that many of the rules will create a conflict with third-party libraries and legacy code (see wordpress markup structure and corysimmons/typographic). You will have to use the control statements to disable the linter accordingly.

**Comment by nuragic** [comment]
Ok, thanks for the clarification @FranzSkuffka !
